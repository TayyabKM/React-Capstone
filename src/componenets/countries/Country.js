import React from 'react';
import PropTypes from 'prop-types';
import '../../assets/country.css';
import { useNavigate } from 'react-router-dom';
import next from '../../assets/images/next.png';

function Country(props) {
  const navigate = useNavigate();

  const {
    id, name, capital, population,
  } = props;
  const capitalString = Array.isArray(capital) ? capital.join(', ') : capital;

  const getCountry = () => {
    navigate(`/detail-list/${id}`);
  };
  return (
    <>
      <div
        role="button"
        tabIndex={0}
        className="col-6 p-5 colorbg text-white"
        id="paddingMobile"
        onClick={getCountry}
        onKeyDown={(event) => {
          if (event.key === 'Enter' || event.key === ' ') {
            getCountry();
          }
        }}
      >
        <div className="spaceBetween">
          <div>
            <div><h2>{name}</h2></div>
            <div>
              <span>Capital: </span>
              {capitalString}
            </div>
            <div>
              <span>Population: </span>
              {population}
            </div>
          </div>
          <div>
            <div className="icon-container">
              <img src={next} alt="next" />
            </div>
          </div>
        </div>

      </div>
    </>
  );
}

Country.propTypes = {
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  capital: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string),
  ]).isRequired,
  population: PropTypes.number.isRequired,
};

export default Country;
