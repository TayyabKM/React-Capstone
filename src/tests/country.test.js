import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Country from '../componenets/countries/Country';

it('renders correctly', () => {
  const country = {
    id: '1',
    name: 'Test Country',
    capital: 'Test Capital',
    population: 1000000,
  };
  const tree = renderer
    .create(
      <Router>
        <Country
          id={country.id}
          name={country.name}
          capital={country.capital}
          population={country.population}
        />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
