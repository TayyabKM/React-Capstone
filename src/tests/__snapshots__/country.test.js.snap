// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="col-6 p-5 colorbg text-white"
  id="paddingMobile"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="spaceBetween"
  >
    <div>
      <div>
        <h2>
          Test Country
        </h2>
      </div>
      <div>
        <span>
          Capital: 
        </span>
        Test Capital
      </div>
      <div>
        <span>
          Population: 
        </span>
        1000000
      </div>
    </div>
    <div>
      <div
        className="icon-container"
      >
        <img
          alt="next"
          src="next.png"
        />
      </div>
    </div>
  </div>
</div>
`;
